### PR TITLE
Doc: Reflect Apache HttpClient as Default in AWS Integration Page

### PR DESCRIPTION
**Pull request by HonahX**
Fixes: #8296 

In #7119, the default http client of AWS is switched from `url-connection` to `apache`. This PR update part of the doc to reflect the change.

cc: @singhpk234 @jackye1995 @stevenzwu 

Whenever you have a moment, could you kindly take a look at this PR? Thanks in advance!

### CODE REVIEW

File: docs/aws.md [hunk 1]
```diff
@@ -31,16 +31,16 @@ This section describes how to use Iceberg with AWS.
 
 ## Enabling AWS Integration
 
-The `iceberg-aws` module is bundled with Spark and Flink engine runtimes for all versions from `0.11.0` onwards.
+The `iceberg-aws` module is bundled with Spark and Flink engine runtimes for all versions from `0.11.0` onwards.ƒ
```
Comment (amogh-jahagirdar) [inline docs/aws.md:34]: Could you remove the `f` at the end of the line?

Comment (HonahX) [reply docs/aws.md]: oops. Thanks for pointing this out!

**Review by amogh-jahagirdar — APPROVED** [review]
This looks good to me, I'll give it a bit for @jackye1995 and others to review and then I'll merge. Thanks for the contribution!

**Review by singhpk234 — APPROVED** [review]
LGTM as well, thanks @HonahX !

**Comment by amogh-jahagirdar** [comment]
Thanks for reviewing @singhpk234 , merging.

**Pull request merged**